xarray/core/resample.py

```diff
@@ -43,12 +67,14 @@ def _flox_reduce(self, dim, **kwargs):
         labels = np.repeat(self._unique_coord.data, repeats)
         group = DataArray(labels, dims=(self._group_dim,), name=self._unique_coord.name)
 
-        result = super()._flox_reduce(dim=dim, group=group, **kwargs)
+        result = super()._flox_reduce(
+            dim=dim, group=group, keep_attrs=keep_attrs, **kwargs
+        )
         result = self._maybe_restore_empty_groups(result)
         result = result.rename({RESAMPLE_DIM: self._group_dim})
         return result
 
-    def _upsample(self, method, *args, **kwargs):
+    def _upsample(self, method, *args, **kwargs) -> T_Xarray:
```

Comment:
Looking at how `_upsample` is actually used, we don't actually use `*args` and `**kwargs`. We only ever pass in a keyword argument `tolerance`.

So I would suggest simplifying the signature here (e.g., at least removing `*args`) which will allow for a cleaner typing solution.

See my comment below for more details.